lib/ansible/modules/git.py

```diff
@@ -1123,11 +1133,18 @@ def main():
 
     if module.params['accept_hostkey']:
         if ssh_opts is not None:
-            if "-o StrictHostKeyChecking=no" not in ssh_opts:
+            if ("-o StrictHostKeyChecking=no" not in ssh_opts) and ("-o StrictHostKeyChecking=accept-new" not in ssh_opts):
                 ssh_opts += " -o StrictHostKeyChecking=no"
         else:
             ssh_opts = "-o StrictHostKeyChecking=no"
 
+    if module.params['accept_newhostkey']:
+        if ssh_opts is not None:
+            if ("-o StrictHostKeyChecking=no" not in ssh_opts) and ("-o StrictHostKeyChecking=accept-new" not in ssh_opts):
```

Comment:
Since this option requires OpenSSH >= 7.5 as you mentioned in the PR description, we should either feature probe by running `ssh -o StrictHostKeyChecking=accept-new` and looking for 'unsupported option' in `stderr`, or check the OpenSSH version before adding the option.

We do not have OpenSSH version information stored anywhere, so adding an `_ssh_version()` function that returns a `LooseVersion()` of the current OpenSSH version is probably the best way to get the current version, especially since it needs to be the version on the remote node not the control node.

If the OpenSSH version is < 7.5, then we should warn and not add `StrictHostKeyChecking=accept-new` to `ssh_opts`.

Reply:
Actually I tried adding ssh version as a fact a week ago in https://github.com/aminvakil/ansible/commit/36bcd3a9085ee8edec0be417e5baca7ad4383c9e, but as it should work without `gather_facts` too, it doesn't help in this matter, yes?

I will try adding `_ssh_version()` function in `lib/ansible/modules/git.py` then.

Reply:
Modules don't have a way to access facts directly, so it's best if the module does this.

Reply:
I've added `get_ssh_version()` function, so could you please review it?